src/meta/src/stream/stream_graph.rs

```diff
@@ -541,41 +541,40 @@ impl StreamGraphBuilder {
+                        // The operator id must be assigned with table ids. Otherwise it is a logic
+                        // error.
+                        let left_table_id = node.left_table_id + table_id_offset;
+                        let right_table_id = left_table_id + 1;
+                        node.left_table_id = left_table_id;
+                        node.right_table_id = right_table_id;
+                    }
 
-                if let NodeBody::HashJoin(node) = new_stream_node.node_body.as_mut().unwrap() {
-                    // The operator id must be assigned with table ids. Otherwise it is a logic
-                    // error.
-                    let left_table_id = node.left_table_id + table_id_offset;
-                    let right_table_id = left_table_id + 1;
-                    node.left_table_id = left_table_id;
-                    node.right_table_id = right_table_id;
-                }
-
-                if let NodeBody::Lookup(node) = new_stream_node.node_body.as_mut().unwrap() {
-                    if let Some(ArrangementTableId::TableId(table_id)) =
-                        &mut node.arrangement_table_id
-                    {
-                        *table_id += table_id_offset;
+                    NodeBody::Lookup(node) => {
+                        if let Some(ArrangementTableId::TableId(table_id)) =
+                            &mut node.arrangement_table_id
+                        {
+                            *table_id += table_id_offset;
+                        }
                     }
-                }
 
-                if let NodeBody::Arrange(node) = new_stream_node.node_body.as_mut().unwrap() {
-                    node.table_id += table_id_offset;
-                }
+                    NodeBody::Arrange(node) => {
+                        node.table_id += table_id_offset;
+                    }
 
-                if let NodeBody::HashAgg(node) = new_stream_node.node_body.as_mut().unwrap() {
-                    assert_eq!(node.table_ids.len(), node.agg_calls.len());
-                    // In-place update the table id. Convert from local to global.
-                    for table_id in &mut node.table_ids {
-                        *table_id += table_id_offset;
+                    NodeBody::HashAgg(node) => {
+                        assert_eq!(node.table_ids.len(), node.agg_calls.len());
+                        // In-place update the table id. Convert from local to global.
+                        for table_id in &mut node.table_ids {
+                            *table_id += table_id_offset;
+                        }
                     }
-                }
 
-                if let NodeBody::TopN(node) = new_stream_node.node_body.as_mut().unwrap() {
-                    node.table_id += table_id_offset;
-                }
+                    NodeBody::TopN(node) | NodeBody::AppendOnlyTopN(node) => {
+                        node.table_id += table_id_offset;
+                    }
```

Comment:
missing `AppendOnlyTopN` here?

Reply:
NodeBody::TopN(node) | NodeBody::AppendOnlyTopN(node) => xxx

TopN and AppendOnlyTopN use the same table_id assign logic, so they use the same pattern

Reply:
That's cool. I missed that 🤣